### DISCUSSION TITLE
i18n reactivity inside <script setup>

### DISCUSSION THREAD

**Discussion by codingwarriortech** [quasarframework/quasar#15942] [General - Components / Directives / etc]
Hi,

I've configured a language switcher in the header component of the MainLayout.
When the language is changed, the header and footer are updated accordingly.
On a page I'd like to get the translation of 'example' .
When I'm using 

```html
<p>{{ $t('example') }}</p>
```

inside the template, it works fine and got reactivity as expected.

But using it inside <script setup> does not give me the reactivity.
I'd have to revisit the page after I've switched the language. 

```html
<template>
  <p>{{  example  }}</p>
</template>

<script setup>
import { ref } from 'vue'
import { useI18n } from 'vue-i18n'

const { t } = useI18n()

const example = ref(t('example'))
</script>
```

I've tried just by using `const example = t('example')` before, but this is just a non-reactive String.

How can I make it reactive?

Thanks a lot.

**Comment by codingwarriortech** [comment]
Using a computed property works:

`const example = computed(() => t('range'))` 

But what happens here exactly?

**Reply by yusufkandemir** [reply]
`t` is a reactive function that returns a string value. It's a reactive function but the value it returns is static.

What happens is similar to below:
```js
const list = ref([5])
const getLast = () => list.value.at(-1)

const lastElement = ref(getLast() /* assigning a static value in the end */) // 5

list.push(1)

console.log(lastElement.value) // still 5
```

**Comment by nodescan-admin** [comment]
If computed property serves your needs then please continue using that. 
I use watch locale like this to translate without reloading the page.

in (script) setup...
 const { locale, t } = useI18n()
 const translatedText = ref(t(props.text)) // if necessary you can translate the initial prop

 watch(locale, () => {
      // console.log('Watch locale: ' + locale.value)
      translatedText.value = t(props.text)
 })

**Reply by TobyMosque** [reply]
would be better to use a computed property
```ts
const { t } = useI18n()
const props = defineProps<{ text: string }>()
const { text } = toRefs(props)

const localized = computed(() => t(text.value))
```
